src/app/components/SearchBox/SearchBox.tsx

```diff
@@ -1,25 +1,45 @@
 'use client';
 
-import React, { FormEvent, useState } from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { useRouter } from 'next/navigation';
 import Image from 'next/image';
+import useDebounce from '@/hooks/useDebounce';
+import { useLoadingContext } from '@/context/loading.context';
 import styles from './SearchBox.module.scss';
 
 function SearchBox() {
   const [input, setInput] = useState('');
+  const { loading } = useLoadingContext();
   const router = useRouter();
+  const debouncedInput = useDebounce(input, 1000);
 
-  const handleSearch = (e: FormEvent<HTMLFormElement>) => {
-    e.preventDefault();
-    router.push(`?q=${input}`);
-  };
+  const handleSearch = useCallback(() => {
+    if (debouncedInput) {
+      router.push(`?q=${debouncedInput}`);
+    }
+  }, [debouncedInput, router]);
+
+  useEffect(() => {
+    handleSearch();
+  }, [debouncedInput, handleSearch]);
 
   return (
-    <form className={styles.searchBox} onSubmit={handleSearch}>
-      <input type="text" placeholder="Search keywords..." value={input} onChange={(e) => setInput(e.target.value)} />
-      <button type="submit" disabled={!input}>
+    <form
+      className={styles.searchBox}
+      onSubmit={(e) => {
+        e.preventDefault();
+      }}
+    >
+      <input
```

Comment:
<img width="771" alt="Screenshot 2023-10-23 at 20 31 22" src="https://github.com/EugeneTsalko/news-api/assets/47400885/0bfce244-1ddb-430e-9c84-9381ffdc66b5">
1. Please make an input’s disable state good looking because currently a grey rectangle in the middle of the input doesn’t look good
2. Add some styles to show a focus state on the input